### PR TITLE
[MM-43444] Log profile image deletion issues from server

### PR DESCRIPTION
**Pull request by plant99**
#### Summary

This change logs if there were issues while deleting profile image of a user when permanently deleting the account, so the admin can take manual steps.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/19996
Jira: https://mattermost.atlassian.net/browse/MM-43444

#### Related PRs
https://github.com/mattermost/mattermost-server/pull/20033

### CODE REVIEW

File: commands/user.go [hunk 1]
```diff
@@ -649,9 +649,13 @@ func deleteUsersCmdF(c client.Client, cmd *cobra.Command, args []string) error {
...
 			continue
 		}
-		if _, err := deleteUser(c, user); err != nil {
+		if res, err := deleteUser(c, user); err != nil {
 			printer.PrintError("Unable to delete user '" + user.Username + "' error: " + err.Error())
 		} else {
+			// res.StatusCode is checked for 202 to identify issues with file deletion.
+			if res.StatusCode == http.StatusAccepted {
+				printer.PrintT("There were issues with deleting profile image of the user. Please delete it manually. Id: '{{.Id}}'", user)
+			}
```
Comment (mkraft) [inline commands/user.go:658]: I don't think we need `deleteUser`, so if you want to you could probably do...

```diff
diff --git a/commands/user.go b/commands/user.go
index 186060c0..33877f03 100644
--- a/commands/user.go
+++ b/commands/user.go
@@ -628,10 +628,6 @@ func resetUserMfaCmdF(c client.Client, cmd *cobra.Command, args []string) error
 	return nil
 }
 
-func deleteUser(c client.Client, user *model.User) (*model.Response, error) {
-	return c.PermanentDeleteUser(user.Id)
-}
-
 func deleteUsersCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	confirmFlag, _ := cmd.Flags().GetBool("confirm")
 	if !confirmFlag {
@@ -649,7 +645,7 @@ func deleteUsersCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 			printer.PrintError("Unable to find user '" + args[i] + "'")
 			continue
 		}
-		if res, err := deleteUser(c, user); err != nil {
+		if res, err := c.PermanentDeleteUser(user.Id); err != nil {
 			printer.PrintError("Unable to delete user '" + user.Username + "' error: " + err.Error())
 		} else {
 			// res.StatusCode is checked for 202 to identify issues with file deletion.
```

0/5 if you want to.

**Review by mkraft — CHANGES_REQUESTED** [review]
Could you please add a test to commands/user_test.go that mocks the case of a 202 status and then ensure that you get the expected warning printed?

**Review by mkraft — APPROVED** [review]

**Comment by plant99** [comment]
Hi @mgdelacroix gentle ping to review this. I'm sure it might have slipped out of your notice since it's been a while, and is of less priority. :)

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1: Apologies for the delay on the review!

**Comment by plant99** [comment]
> LGTM! 👍 Apologies for the delay on the review!

No worries at all, thanks a lot \o/

**Pull request merged**